nxdrive/gui/api.py

```diff
@@ -322,19 +321,17 @@ def app_update(self, version: str) -> None:
         """ Start the udpate to the specified version. """
         self._manager.updater.update(version)
 
-    @pyqtSlot(str, result=str)
-    def get_actions(self, uid: str) -> str:
-        engine = self._get_engine(uid)
+    @pyqtSlot(result=list)
+    def get_actions(self) -> List[Dict[str, Any]]:
         result = []
-        if engine:
-            for count, thread in enumerate(engine.get_threads(), 1):
-                action = thread.worker.action
-                # The filter should be configurable
-                if isinstance(action, FileAction):
-                    result.append(self._export_action(action))
-                if count == 4:
-                    break
-        return self._json(result)
+
+        if not self._manager.get_engines():
+            return result
+
+        for thread, action in Action.actions.copy().items():
```

Comment:
`thread` is unused, perhasp it is better to use `Action.actions.copy().values()`?